CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -267,7 +267,7 @@ private function getNonDeductibleAmount($params, $financialType, $online) {
    */
   public function preProcess() {
     parent::preProcess();
-    $this->_ccid = $this->get('ccid');
+    $this->_ccid = $this->getExistingContributionID();
```

Comment:
I've put merge-ready because while in a quick test it seems to run ok, I'm not sure this is a drop-in replacement for the get() calls (get retrieves it from the session).

It does seem like a drop-in though for the replacements of `$this->_ccid`.

Reply:
You might also set `_ccid` inside of `getExistingContributionID()`

Reply:
(ultimately it could be made protected & then we used a `__get()` function to return it with a deprecation notice if people try to access directly)